File: app/src/main/java/com/summerland/android/game/Smoke.java
package com.summerland.android.game;

import android.graphics.Canvas;
import android.graphics.Color;
import android.graphics.Paint;

/**
 * Smoke class, Created by steve on 01/10/16.
 */
public class Smoke extends GameObject {

    public static final int DIA = 4;
    public static final int UP_DIA = 5;

    public Smoke(int x, int y){
        this.x = x;
        this.y = y;
    }

    @Override
    public void update() {
        x -= 10;
    }

    @Override
    public void draw(Canvas canvas) {
        draw(canvas, true);
    }

    public void draw(Canvas canvas, boolean goingUp) {

        int size = DIA;
        Paint paint = new Paint();
        paint.setColor(Color.GRAY);
        paint.setStyle(Paint.Style.FILL);

        if (goingUp) {
            paint.setColor(Color.DKGRAY);
            size = UP_DIA;
        }

        canvas.drawCircle(x - size, y - size, size, paint);
        canvas.drawCircle(x - size + 2, y - size - 2, size, paint);
        paint.setColor(Color.GRAY);
        canvas.drawCircle(x - size + 4, y - size - 2, size, paint);
    }
}

File: app/src/main/java/com/summerland/android/game/Smoke.kt
package com.summerland.android.game

import android.graphics.Canvas
import android.graphics.Color
import android.graphics.Paint

/**
 * Smoke class, Created by steve on 01/10/16.
 */
class Smoke internal constructor(x: Int, y: Int) : GameObject() {

    init {
        this.x = x
        this.y = y
    }

    override fun update() {
        x -= 10
    }

    override fun draw(canvas: Canvas) {
        draw(canvas, true)
    }

    fun draw(canvas: Canvas, goingUp: Boolean) {

        var size = DIA
        val paint = Paint()
        paint.color = Color.GRAY
        paint.style = Paint.Style.FILL

        if (goingUp) {
            paint.color = Color.DKGRAY
            size = UP_DIA
        }

        canvas.drawCircle((x - size).toFloat(), (y - size).toFloat(), size.toFloat(), paint)
        canvas.drawCircle((x - size + 2).toFloat(), (y - size - 2).toFloat(), size.toFloat(), paint)
        paint.color = Color.GRAY
        canvas.drawCircle((x - size + 4).toFloat(), (y - size - 2).toFloat(), size.toFloat(), paint)
    }

    companion object {
        private val DIA = 4
        private val UP_DIA = 5
    }
}
